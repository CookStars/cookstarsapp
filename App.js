import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { NavigationContainer, StackActions } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

function HomeScreen() {
  return (
    <View style={styles.container}>
      <Text>Home Screen</Text>
    </View>
  );
}

function SecondScreen() {
  return (
    <View style={styles.container}>
      <Text>Second Screen</Text>
    </View>
  );
}

const Stack = createStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName='Home'>
        <Stack.Screen name='Home' component={HomeScreen} />
        <Stack.Screen name='Second' component={SecondScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
