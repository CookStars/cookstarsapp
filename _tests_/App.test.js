import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'

describe('<App />', () => {
    it('App renders without crashing', () => {
        const rendered = renderer.create(<App />).toJSON()
        expect(rendered).toBeTruthy()
    })

    it('App test against snapshot', () => {
        const tree = renderer.create(<App />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('has 1 child', () => {
        const tree = renderer.create(<App />).toJSON()
        expect(tree.children.length).toBe(1)
    })
})
