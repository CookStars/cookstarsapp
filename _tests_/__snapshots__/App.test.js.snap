// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          />
                          <View
                            accessibilityElementsHidden={false}
                            closing={false}
                            gestureVelocityImpact={0.3}
                            importantForAccessibility="auto"
                            onClose={[Function]}
                            onGestureBegin={[Function]}
                            onGestureCanceled={[Function]}
                            onOpen={[Function]}
                            onTransitionStart={[Function]}
                            pointerEvents="box-none"
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                            transitionSpec={
                              Object {
                                "close": Object {
                                  "animation": "spring",
                                  "config": Object {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 10,
                                    "restSpeedThreshold": 10,
                                    "stiffness": 1000,
                                  },
                                },
                                "open": Object {
                                  "animation": "spring",
                                  "config": Object {
                                    "damping": 500,
                                    "mass": 3,
                                    "overshootClamping": true,
                                    "restDisplacementThreshold": 10,
                                    "restSpeedThreshold": 10,
                                    "stiffness": 1000,
                                  },
                                },
                              }
                            }
                          >
                            <View
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "flex": 1,
                                  "marginTop": 0,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                forwardedRef={[Function]}
                                onGestureHandlerEvent={[Function]}
                                onGestureHandlerStateChange={[Function]}
                                style={
                                  Object {
                                    "flex": 1,
                                    "transform": Array [
                                      Object {
                                        "translateX": 0,
                                      },
                                      Object {
                                        "translateX": 0,
                                      },
                                    ],
                                  }
                                }
                              >
                                <View
                                  pointerEvents="none"
                                  style={
                                    Object {
                                      "backgroundColor": "rgb(242, 242, 242)",
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "shadowColor": "#000",
                                      "shadowOffset": Object {
                                        "height": 1,
                                        "width": -1,
                                      },
                                      "shadowOpacity": 0.3,
                                      "shadowRadius": 5,
                                      "top": 0,
                                      "width": 3,
                                    }
                                  }
                                />
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "flex": 1,
                                        "overflow": "hidden",
                                      },
                                      Array [
                                        Object {
                                          "backgroundColor": "rgb(242, 242, 242)",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "flexDirection": "column-reverse",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "flex": 1,
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#F4F1DE",
                                            "flex": 1,
                                            "justifyContent": "center",
                                          }
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "marginBottom": 20,
                                              "marginTop": 10,
                                            }
                                          }
                                        />
                                        <View />
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "backgroundColor": "#FFFFFF",
                                              "borderBottomColor": "#F2CC8F",
                                              "borderBottomWidth": 1,
                                              "borderRadius": 30,
                                              "flexDirection": "row",
                                              "height": 45,
                                              "marginBottom": 20,
                                              "width": 250,
                                            }
                                          }
                                        >
                                          <TextInput
                                            allowFontScaling={true}
                                            id="emailInput"
                                            keyboardType="email-address"
                                            onChangeText={[Function]}
                                            placeholder="Email"
                                            rejectResponderTermination={true}
                                            style={
                                              Object {
                                                "borderBottomColor": "#F2CC8F",
                                                "flex": 1,
                                                "height": 45,
                                                "marginLeft": 16,
                                              }
                                            }
                                            underlineColorAndroid="transparent"
                                            value=""
                                          />
                                        </View>
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "backgroundColor": "#FFFFFF",
                                              "borderBottomColor": "#F2CC8F",
                                              "borderBottomWidth": 1,
                                              "borderRadius": 30,
                                              "flexDirection": "row",
                                              "height": 45,
                                              "marginBottom": 20,
                                              "width": 250,
                                            }
                                          }
                                        >
                                          <TextInput
                                            allowFontScaling={true}
                                            onChangeText={[Function]}
                                            placeholder="Password"
                                            rejectResponderTermination={true}
                                            secureTextEntry={true}
                                            style={
                                              Object {
                                                "borderBottomColor": "#F2CC8F",
                                                "flex": 1,
                                                "height": 45,
                                                "marginLeft": 16,
                                              }
                                            }
                                            underlineColorAndroid="transparent"
                                            value=""
                                          />
                                        </View>
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Array [
                                              Object {
                                                "alignItems": "center",
                                                "borderRadius": 30,
                                                "flexDirection": "row",
                                                "height": 45,
                                                "justifyContent": "center",
                                                "marginBottom": 20,
                                                "width": 250,
                                              },
                                              Object {
                                                "backgroundColor": "#F18F01",
                                              },
                                            ]
                                          }
                                        >
                                          <Text
                                            style={
                                              Object {
                                                "color": "white",
                                              }
                                            }
                                          >
                                            Login
                                          </Text>
                                        </View>
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "borderRadius": 30,
                                              "flexDirection": "row",
                                              "height": 45,
                                              "justifyContent": "center",
                                              "marginBottom": 20,
                                              "width": 250,
                                            }
                                          }
                                        >
                                          <Text>
                                            Forgot your password?
                                          </Text>
                                        </View>
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "borderRadius": 30,
                                              "flexDirection": "row",
                                              "height": 45,
                                              "justifyContent": "center",
                                              "marginBottom": 20,
                                              "width": 250,
                                            }
                                          }
                                        >
                                          <Text>
                                            Register
                                          </Text>
                                        </View>
                                        <ActivityIndicator
                                          animating={true}
                                          color="#999999"
                                          hidesWhenStopped={true}
                                          size="large"
                                        />
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        pointerEvents="box-none"
                        style={
                          Array [
                            Object {
                              "zIndex": 1,
                            },
                            Object {
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
